File: Java/DemoProject/src/main/java/lab/lab10.java
package lab;
abstract class Animal {
 abstract void makeSound();
}

class Dog extends Animal {
 void makeSound() {
     System.out.println("dog makes Woof sound");
 }
}

class Cat extends Animal {
 void makeSound() {
     System.out.println("cat makes Meow sound");
 }
}

public class lab10 {
 public static void main(String[] args) {
     Animal dog = new Dog();
     Animal cat = new Cat();

     dog.makeSound();
     cat.makeSound();
     System.out.println("Copyright Pitam Poudel");
 }
}

File: Web/Sem 4 Lab/lab10.php
<?php
// LAB 10: Implementing Inheritance in PHP
echo "<h2>LAB 10: Implementing Inheritance in PHP</h2>";

// Parent class
class Animal
{
    public function speak()
    {
        return "animal makes a sound<br>";
    }
}

// Child class inheriting from Animal
class Dog extends Animal
{
    public function speak()
    {
        return "dog barks<br>";
    }
}

// Create objects
$dog = new Dog();

echo $dog->speak();

echo "<br>© 2025 Pitam. All rights reserved.";
